### PR TITLE
Cloud Native Security Whitepaper submitted

### PR DESCRIPTION
**Pull request by idvoretskyi**
Signed-off-by: Ihor Dvoretskyi <ihor@linux.com>

### CODE REVIEW

**Comment by ultrasaurus** [comment]
I haven't looked through content, since I know you are still working on it.  Just wanted to make a couple of requests:
- can you make filenames lowercase using dashes to separate words?  I don't actually see that in CONTRIBUTING.md (which I will add) -- it's pretty consistent across the repo 
- can you please check in the source to the images?  I assume we'll have future versions and this will be the canonical source files for future versions

Thanks so much for your work on the conversion!  We should pull in White Paper authors when it is ready for review.

File: security-whitepaper/cloud_native_security_whitepaper.md [hunk 1]
```diff
@@ -0,0 +1,845 @@
+Shared with CNCF Community
+
+# **Cloud Native Security Whitepaper**
+
+|
+**Work-Group:** SIG-Security **Short self link:** https://tinyurl.com/CNCF-CNSWP | **Status** : WIP | **In Review** | Approved | Obsolete
```
Comment (TheFoxAtWork) [inline security-whitepaper/cloud_native_security_whitepaper.md:6]:
```suggestion
**Work-Group:** SIG-Security **Short self link:** https://tinyurl.com/CNCF-CNSWP | **Status** : WIP | In Review | **Approved** | Obsolete
```

File: security-whitepaper/cloud_native_security_whitepaper.md [hunk 2]
```diff
@@ -0,0 +1,845 @@
...
+**Created** : 2020-FEB-01 **Review By** : 2020-OCT-27 **Release Version:** 0.0.1 **Approvers** : [x] |
```
Comment (TheFoxAtWork) [inline security-whitepaper/cloud_native_security_whitepaper.md:7]:
```suggestion
**Created** : 2020-FEB-01 **Reviewed** : 2020-OCT-27  **Published** : 2020-NOV-18 **Release Version:** 1.0 **Approvers** : [x] @lizrice [x] @justincormack  |
```

**Comment by TheFoxAtWork** [comment]
@idvoretskyi a quick skim - everything looks in place.  I made some alterations to the initial content in preparation for the upcoming post.

**Comment by idvoretskyi** [comment]
@TheFoxAtWork thanks!

File: security-whitepaper/cloud-native-security-whitepaper.md [hunk 3]
```diff
@@ -0,0 +1,846 @@
...
+- @craigbox
+- @anvega
+- @magnologan
+- Alok Raj - [XenonStack](https://www.xenonstack.com/) ([alok@xenonstack.com](mailto:alok@xenonstack.com))
+- @nyrahul
+- @ranio1
+
+[1](#sdfootnote1anc) Another model to consider is Cloud, Clusters, Containers, and Code: [https://kubernetes.io/docs/concepts/security/overview/](https://kubernetes.io/docs/concepts/security/overview/)
+
+[2](#sdfootnote2anc)[Shifting security left](https://www.devsecops.org/blog/2016/5/20/-security) often leaves organizations to lapse operational security monitoring. It is important that security exists in all parts of the lifecycle and organizations continually evaluate other aspects of their business and technology processes where they may reach beyond modern security paradigms to embrace security as a culture and habit.
```
Comment (PushkarJ) [inline security-whitepaper/cloud-native-security-whitepaper.md:830]: Needs a space like this:
`[2](#sdfootnote2anc) [Shifting`

File: security-whitepaper/cloud-native-security-whitepaper.md [hunk 4]
```diff
@@ -0,0 +1,846 @@
...
+
+[3] Example - [MITRE ATT&CK Framework for Kubernetes](https://www.darkreading.com/threat-intelligence/microsofts-kubernetes-threat-matrix-heres-whats-missing/a/d-id/1339106)
+
+[3](#sdfootnote3anc) Human capital is a vital asset necessary to the success of any organization, the corresponding intellectual property and relational capital brought as a result is equally in need of protection.
+
+[4](#sdfootnote4anc) https://blog.aquasec.com/malicious-container-image-docker-container-host
+
+[5](#sdfootnote5anc) According to Applied Software Measurement, Capers Jones, 1996 and adjusting for inflation - 85% of defects are introduced during coding with a cost of $41 to fix compared to a post release fix cost of $26,542.
+
+[6](#sdfootnote6anc) CIsecurity.org maintains a listing of benchmarks for hardening
```
Comment (PushkarJ) [inline security-whitepaper/cloud-native-security-whitepaper.md:840]: This should be cisecurity.org without upper case as it does _not_ represent CI (Continuous Integration).

File: security-whitepaper/cloud-native-security-whitepaper.md [hunk 5]
```diff
@@ -0,0 +1,846 @@
+Shared with CNCF Community
+
+# **Cloud Native Security Whitepaper**
+
+|
+**Work-Group:** SIG-Security **Short self link to working doc:** https://tinyurl.com/CNCF-CNSWP | **Status** : WIP | In Review | **Approved** | Obsolete
+
+**Created** : 2020-FEB-01 **Reviewed** : 2020-OCT-27  **Published** : 2020-NOV-18 **Release Version:** 1.0 **Approvers** : [x] @lizrice [x] @justincormack  |
+| **Contributors:** [aradhna.chetal@gmail.com](mailto:aradhna.chetal@gmail.com), [themoxiefoxatwork@gmail.com](mailto:themoxiefoxatwork@gmail.com), [jj@tetrate.io](mailto:jj@tetrate.io), gadi@alcide.io @lumjjb, @trishankatdatadog, [@vvenkatara@paloaltonetworks.com](mailto:vvenkatara@paloaltonetworks.com), @pushkarj-v, @whaber, @sublimino, @rowan-baker, [chase.pettet@gmail.com](mailto:chase.pettet@gmail.com), [harsingh@us.ibm.com](mailto:harsingh@us.ibm.com), jeff.lombardo@gmail.com
```
Comment (PushkarJ) [inline security-whitepaper/cloud-native-security-whitepaper.md:9]: @idvoretskyi : Had a recent change in my Github ID/username and the one here does not exist. Can you please replace it with `@pushkarj-v` with `@pushkarj`. Thanks and sorry for the bother.

File: security-whitepaper/cloud-native-security-whitepaper.md [hunk 6]
```diff
@@ -0,0 +1,846 @@
...
+
+- Image signature and integrity
+- Image runtime policies (e.g absence of malware or critical vulnerabilities)
+- Container runtime policies (e.g absence of excessive privileges)
+- Host vulnerability and compliance controls
+- Workload, application, and network security policies
+
+#### Observability & Metrics
+
+Instituting observability and metrics into cloud native architectures delivers security insights so appropriate stakeholders can resolve and mitigate anomalies appearing in reporting; tools in this area can help collect and visualize this information. Through the use of behavioral and heuristic analysis teams can detect and escalate outliers, suspicious events, and unexplained calls to appropriate stakeholders. Artificial intelligence (AI), [machine learning (ML)](#kix.tlhq6ssgript), or statistical modeling are all mechanisms that are encouraged to assist in behavioral and heuristic analysis development.
```
Comment (PushkarJ) [inline security-whitepaper/cloud-native-security-whitepaper.md:371]: Is the referenced link for ML intentional ?

File: security-whitepaper/cloud-native-security-whitepaper.md [hunk 7]
```diff
@@ -0,0 +1,846 @@
...
+
+It is important to note that implementation of a [service mesh](#_8999rr470gwb) can help reduce the attack surface of a cloud native deployment, and provide a key framework for building [zero trust](#kix.kyffcmw3zhk)application networks.
```
Comment (PushkarJ) [inline security-whitepaper/cloud-native-security-whitepaper.md:454]: Missing space between `zero trust` and `application`

File: security-whitepaper/cloud-native-security-whitepaper.md [hunk 8]
```diff
@@ -0,0 +1,846 @@
...
+- @magnologan
+- Alok Raj - [XenonStack](https://www.xenonstack.com/) ([alok@xenonstack.com](mailto:alok@xenonstack.com))
+- @nyrahul
+- @ranio1
+
+[1](#sdfootnote1anc) Another model to consider is Cloud, Clusters, Containers, and Code: [https://kubernetes.io/docs/concepts/security/overview/](https://kubernetes.io/docs/concepts/security/overview/)
+
+[2](#sdfootnote2anc)[Shifting security left](https://www.devsecops.org/blog/2016/5/20/-security) often leaves organizations to lapse operational security monitoring. It is important that security exists in all parts of the lifecycle and organizations continually evaluate other aspects of their business and technology processes where they may reach beyond modern security paradigms to embrace security as a culture and habit.
+
+[3] Example - [MITRE ATT&CK Framework for Kubernetes](https://www.darkreading.com/threat-intelligence/microsofts-kubernetes-threat-matrix-heres-whats-missing/a/d-id/1339106)
```
Comment (PushkarJ) [inline security-whitepaper/cloud-native-security-whitepaper.md:832]: possible numbering issue because of duplicate 3

**Comment by PushkarJ** [comment]
@idvoretskyi added a few comments, mostly _nits_. Thank you so much, for your excellent work :)

File: security-whitepaper/cloud-native-security-whitepaper.md (hunk 5 above)
Comment (lumjjb) [reply security-whitepaper/cloud-native-security-whitepaper.md:9]: Can we add links to the github IDs as well?

**Comment by idvoretskyi** [comment]
@PushkarJ thanks a lot, will address them!

File: security-whitepaper/cloud-native-security-whitepaper.md (hunk 6 above)
Comment (idvoretskyi) [reply security-whitepaper/cloud-native-security-whitepaper.md:371]: @PushkarJ it's not - will be fixed.

File: security-whitepaper/cloud-native-security-whitepaper.md (hunk 8 above)
Comment (idvoretskyi) [reply security-whitepaper/cloud-native-security-whitepaper.md:832]: The same footnote numbering is in the original document as well. @TheFoxAtWork can you please check?

File: security-whitepaper/cloud-native-security-whitepaper.md [hunk 9]
```diff
@@ -0,0 +1,848 @@
...
...rapidly changing environments, prone to fail-forward for iteration, automation inline with the development pipeline is required for secure outcomes. Organizations must earnestly analyze and apply these core security concepts to alleviate delay in applying hardening and environmental controls, and need to hold engaged third parties to the same standard while balancing perpetual education and training relevant to the cloud capabilities and security proponents for their own workforce.With additional layers of complexity and a broad mesh of components to care for, protection from unauthorized access must be accomplished by integrating security throughout the lifecycle and into the runtime environment. It is highly recommended organizations evaluate the security defense stack against the relevant attack frameworks [^3] to achieve clarity about which threats a defense stack covers. Additionally, organizations need to adopt approaches and methodologies that shift security left [^2], amplify DevOps, and reach beyond to the next technology horizon so continued, proper checks of all components before, within, and after the pipeline are verified with any innovation brought into the lifecycle.
```
Comment (TheFoxAtWork) [inline security-whitepaper/cloud-native-security-whitepaper.md:162]:
```suggestion
Cloud native security seeks to ensure the same conditions of diligence, integrity, trust, and threat prevention as traditional security models while integrating modern concepts of ephemerality, distribution, and immutability. In these rapidly changing environments, prone to fail-forward for iteration, automation inline with the development pipeline is required for secure outcomes. Organizations must earnestly analyze and apply these core security concepts to alleviate delay in applying hardening and environmental controls, and need to hold engaged third parties to the same standard while balancing perpetual education and training relevant to the cloud capabilities and security proponents for their own workforce.With additional layers of complexity and a broad mesh of components to care for, protection from unauthorized access must be accomplished by integrating security throughout the lifecycle and into the runtime environment. It is highly recommended organizations evaluate the security defense stack against the relevant attack frameworks [^2] to achieve clarity about which threats a defense stack covers. Additionally, organizations need to adopt approaches and methodologies that shift security left [^3], amplify DevOps, and reach beyond to the next technology horizon so continued, proper checks of all components before, within, and after the pipeline are verified with any innovation brought into the lifecycle.
```

File: security-whitepaper/cloud-native-security-whitepaper.md [hunk 10]
```diff
@@ -0,0 +1,848 @@
...
+- @magnologan
+- Alok Raj - [XenonStack](https://www.xenonstack.com/) ([alok@xenonstack.com](mailto:alok@xenonstack.com))
+- @nyrahul
+- @ranio1
+
+[1^]: Another model to consider is Cloud, Clusters, Containers, and Code: [https://kubernetes.io/docs/concepts/security/overview/](https://kubernetes.io/docs/concepts/security/overview/)
+
+[2^]: [Shifting security left](https://www.devsecops.org/blog/2016/5/20/-security) often leaves organizations to lapse operational security monitoring. It is important that security exists in all parts of the lifecycle and organizations continually evaluate other aspects of their business and technology processes where they may reach beyond modern security paradigms to embrace security as a culture and habit.
+
+[3^]: Example - [MITRE ATT&CK Framework for Kubernetes](https://www.darkreading.com/threat-intelligence/microsofts-kubernetes-threat-matrix-heres-whats-missing/a/d-id/1339106)
```
Comment (TheFoxAtWork) [inline security-whitepaper/cloud-native-security-whitepaper.md:834]:
```suggestion
[2^]: Example - [MITRE ATT&CK Framework for Kubernetes](https://www.darkreading.com/threat-intelligence/microsofts-kubernetes-threat-matrix-heres-whats-missing/a/d-id/1339106)

[3^]: [Shifting security left](https://www.devsecops.org/blog/2016/5/20/-security) often leaves organizations to lapse operational security monitoring. It is important that security exists in all parts of the lifecycle and organizations continually evaluate other aspects of their business and technology processes where they may reach beyond modern security paradigms to embrace security as a culture and habit.
```

**Review by TheFoxAtWork — APPROVED** [review]

File: security-whitepaper/cloud-native-security-whitepaper.md [hunk 11]
```diff
@@ -0,0 +1,848 @@
...
+
+###### Orchestration
+
+Most orchestrated systems will implement a variety of abstraction and virtualization layers that may include filesystems (such as bind mounts), volume managers, and the application of permissions at a user or group level based on orchestrator policies. As with many components of containerization and microservice architectures, protecting volumes and storage will always rely on the protections in place from other capabilities. If a user is able to escalate their privileges within the orchestrator or container runtime to root they can wreak havoc within the environment. The implementation of [zero trust](https://docs.google.com/document/d/1aAkuhZMJ6-Xq6xP0gh1wxIM8OhUETqEFLhoRdxYHN5c/edit#bookmark=kix.kyffcmw3zhk), [least privilege](https://docs.google.com/document/d/1aAkuhZMJ6-Xq6xP0gh1wxIM8OhUETqEFLhoRdxYHN5c/edit#bookmark=id.bl312cp5xyv1), and [access control](https://docs.google.com/document/d/1aAkuhZMJ6-Xq6xP0gh1wxIM8OhUETqEFLhoRdxYHN5c/edit#bookmark=kix.3f703zlu80we) and enforcement are linchpins in successfully securing storage in cloud native architectures.
```
Comment (lumjjb) [inline security-whitepaper/cloud-native-security-whitepaper.md:492]:
```suggestion
Most orchestrated systems will implement a variety of abstraction and virtualization layers that may include filesystems (such as bind mounts), volume managers, and the application of permissions at a user or group level based on orchestrator policies. As with many components of containerization and microservice architectures, protecting volumes and storage will always rely on the protections in place from other capabilities. If a user is able to escalate their privileges within the orchestrator or container runtime to root they can wreak havoc within the environment. The implementation of [zero trust](#zero-trust-architecture), [least privilege](#least-privilege), and [access control](#access) and enforcement are linchpins in successfully securing storage in cloud native architectures.
```

File: security-whitepaper/cloud-native-security-whitepaper.md [hunk 12]
```diff
@@ -0,0 +1,848 @@
...
+These templates should be scanned for insecure configurations and other security controls using automated tools before deploying the application and infrastructure artifacts. Key misconfigurations to keep an eye out for include:
+
+- Vulnerabilities contained within images specified in the application manifests
+- Configuration settings, such as containers that can escalate privileges.
+- Identification of the security contexts and system calls, which can compromise a system.
+- Resource limit settings
+
+##### Dynamic Analysis
+
+Dynamic analysis of deployed infrastructure may include detecting RBAC and IAM configuration drift, validating the expected network attack surface, and ensuring that a SOC can detect unusual behavior in dedicated test environments to configure alerting for production. Dynamic analysis is considered to be a part of testing however, it is expected to occur in a non-production runtime environment.
```
Comment (lumjjb) [inline security-whitepaper/cloud-native-security-whitepaper.md:323]:
```suggestion
Dynamic analysis of deployed infrastructure may include detecting Role-based Access Control (RBAC) and IAM configuration drift, validating the expected network attack surface, and ensuring that a SOC can detect unusual behavior in dedicated test environments to configure alerting for production. Dynamic analysis is considered to be a part of testing however, it is expected to occur in a non-production runtime environment.
```
RBAC first use

**Review by lumjjb — COMMENTED** [review]
Couple suggested changes to remove implementation and vendor specific details, and content rewording, and several other formatting and clarification additions.

File: security-whitepaper/cloud-native-security-whitepaper.md [hunk 13]
```diff
@@ -0,0 +1,848 @@
...
+
+- Encryption with an external Key Management Store (KMS)
+  - Leveraging a KMS is a secure way to protect secrets in the orchestrator secret store where key encryption in an external KMS encrypts the Data Encryption Key (DEK) that encrypts the secrets stored at rest in etcd. This method does have an option to cache DEKs in memory to reduce the dependency on the availability of the external KMS and faster decryption of secrets during pod creation time.
+- Encryption with the key in a config file
+  - This methodology encrypts the secrets stored in the orchestrator, but storage of the encryption key is available in a configuration file on manager nodes.
+- No encryption
+  - For example, with some orchestrators, secrets are base64 encoded and stored in clear-text in the key-value store by default
+
+Using an external secrets manager can limit the risks of using unencrypted secrets and ease the complexity of key management. Most of the time those tools are provided as controllers or operators for Kubernetes that can inject secrets at runtime and handle their rotations transparently.
```
Comment (lumjjb) [inline security-whitepaper/cloud-native-security-whitepaper.md:436]:
```suggestion
Using an external secrets manager can limit the risks of using unencrypted secrets and ease the complexity of key management. Most of the time those tools are provided as controllers or operators that can inject secrets at runtime and handle their rotations transparently.
```

**Comment by idvoretskyi** [comment]
@TheFoxAtWork ping for the final review :)

**Review by TheFoxAtWork — APPROVED** [review]
@lumjjb @pragashj @ultrasaurus  @JustinCappos need a second approval on this to merge

File: security-whitepaper/cloud-native-security-whitepaper.md [hunk 14]
```diff
@@ -0,0 +1,847 @@
...
+- @justincappos
+- @lumjjb
+- @whaber
+- @craigbox
+- @anvega
+- @magnologan
+- Alok Raj - [XenonStack](https://www.xenonstack.com/) ([alok@xenonstack.com](mailto:alok@xenonstack.com))
+- @nyrahul
+- @ranio1
+
```
Comment (TheFoxAtWork) [inline security-whitepaper/cloud-native-security-whitepaper.md:828]:
```suggestion
- @lizrice 
- @justincormack 

```

**Review by lumjjb — APPROVED** [review]
LGTM!

**Review by soapland-master69 — COMMENTED** [review]
micro stack issues

**Pull request merged**